gpytorch/lazy/lazy_tensor.py

```diff
@@ -1609,6 +1594,11 @@ def root_decomposition(self, method: Optional[str] = None):
         from .chol_lazy_tensor import CholLazyTensor
         from .root_lazy_tensor import RootLazyTensor
 
+        # # no need to do extra work if we already have a cached cholesky decomposition
+        # if _is_in_cache_ignore_args(self, "cholesky"):
+        #     res = self.cholesky()
+        #     return CholLazyTensor(res)
```

Comment:
this is sth that I think makes sense. It currently causes some issues though w/ handling `DiagLazyTensors` as proper special cases of `TriangularLazyTensor`. Will have to revisit this some other time.